src/neural/cuda/network_cuda.cc

```diff
@@ -435,165 +438,214 @@ class CudaNetwork : public Network {
 
     DataType* tensor_mem[3];
     void* scratch_mem;
-    cudaStream_t stream;
+    cudaStream_t stream, stream_copy;
     cublasHandle_t cublas;
+    bool use_cuda_graphs = use_cuda_graphs_;
+    bool graph_created = false;
     if (multi_stream_) {
       // We use tensor and scratch memory from InputOutputs (so that multiple
       // requests can run in parallel)
       for (int i = 0; i < 3; i++) tensor_mem[i] = (DataType*)io->tensor_mem_[i];
       scratch_mem = io->scratch_mem_;
       stream = io->stream_;
+      stream_copy = use_async_policy_copy_ ? io->stream_copy_ : stream;
       cublas = io->cublas_;
+      graph_created = io->graph_created_[batchSize];
     } else {
       for (int i = 0; i < 3; i++) tensor_mem[i] = tensor_mem_[i];
       scratch_mem = scratch_mem_;
       stream = 0;           // default stream
+      stream_copy = 0;
+      use_cuda_graphs = false;
       cublas = cublas_;
     }
 
-    bool fp16 = std::is_same<half, DataType>::value;
-    if (fp16) {
-      expandPlanes_Fp16_NCHW((half*)(tensor_mem[0]), ipDataMasks,
+    if (!graph_created) {
+
+      if (use_cuda_graphs)
+        ReportCUDAErrors(cudaStreamBeginCapture(
+            stream, cudaStreamCaptureModeThreadLocal));
```

Comment:
The second argument for `cudaStreamBeginCapture()` was added in cuda 10.1 (while we still use cuda 10.0 for the cudnn builds). Does it make sense to make this conditional? We can update the cuda version used for the builds, but I'm not confident there are no downsides for older gpu support.